### PR TITLE
Remove GPURender code, re-fix some changes that were lost

### PR DESCRIPTION
**Pull request by DamianMadden**
Resolves [AB#419](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/419)
Resolves [AB#3](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/3)

### CODE REVIEW

**Comment by DamianMadden** [comment]
Quick overview: most changes are removing unused GPU render code, shader initialisation is now tracked explicitly because some shaders never have constant buffers after recent changes, some changes got pulled in from other branches for some reason, including stuff I thought already was merged so maybe check with Pete regarding the attribute stuff in scenelayer before pushing this through. The samplers being mostly removed as their own concept outside of vcTexture.mm I thought was already merged, but I guess the changes were lost, this code is not necessary in metal as textures are created with an associated sampler determined by the properties passed to vctexture_create

File: src/gl/metal/vcRenderShaders.cpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ const char* const g_WaterVertexShader = "waterVertexShader";
 const char* const g_PolygonP3N3UV2FragmentShader = "polygonP3N3UV2FragmentShader";
 const char* const g_PolygonP3N3UV2VertexShader = "polygonP3N3UV2VertexShader";
 
-const char* const g_FlatColour_FragmentShader = "flatColourFragmentShader";
+const char* const g_FlatColour_FragmentShader = "flatColorFragmentShader";
```
Comment (OsiJr) [inline src/gl/metal/vcRenderShaders.cpp]: We usually use Australian English for variable names, any reason for this change?

File: src/vcSceneLayer.cpp [hunk 2]
```diff
@@ -353,6 +353,8 @@ udResult vcSceneLayer_LoadFeatureData(const vcSceneLayer *pSceneLayer, vcSceneLa
 
     pNode->pFeatureData[i].pGeometryLayout = udAllocType(vcVertexLayoutTypes, pNode->pFeatureData[i].geometryLayoutCount, udAF_Zero);
     UD_ERROR_NULL(pNode->pFeatureData[i].pGeometryLayout, udR_MemoryAllocationFailure);
+    pNode->pFeatureData[i].pAttributeOffset = udAllocType(uint64_t, pNode->pFeatureData[i].geometryLayoutCount, udAF_Zero);
```
Comment (OsiJr) [inline src/vcSceneLayer.cpp]: This is unrelated work for the SLPK things?

File: src/gl/metal/vcRenderShaders.cpp (hunk 1 above)
Comment (DamianMadden) [reply src/gl/metal/vcRenderShaders.cpp]: Well no reason specifically that I chose american english, but the change was just for uniformity

File: src/vcSceneLayer.cpp (hunk 2 above)
Comment (DamianMadden) [reply src/vcSceneLayer.cpp]: Yes, I'm sort of confused as to how that got into this PR, I thought it had been merged already, I'm guessing it was in master at some point and then I started working on top of that branch... Then maybe pete dumped the attribute offset change? He mentioned to me on Tuesday that the other change I made related to that was the actual fix and this change didn't contribute but nonetheless was correct and should be done

**Comment by OsiJr** [comment]
Can you rebase this onto latest so we can rule out weird git changes

**Comment by samsinsane** [comment]
You've rebased or merged your branch with the wrong target branch, you're merging into `master-vdk040` not `master`.

File: src/vcRender.cpp [hunk 3]
```diff
@@ -1193,15 +1192,14 @@ vcRenderPickResult vcRender_PolygonPick(vcState *pProgramState, vcRenderContext
...
+#if GRAPHICS_API_OPENGL || GRAPHICS_API_METAL
     pickDepth = uint32_t((depthBytes[3] << 16) | (depthBytes[2] << 8) | (depthBytes[1] << 0)) / ((1 << 24) - 1.0f);
     //uint8_t stencil = depthBytes[0];
 #else
-  // TODO: byte order of metal etc.
     pickDepth = uint32_t((depthBytes[2] << 16) | (depthBytes[1] << 8) | (depthBytes[0] << 0)) / ((1 << 24) - 1.0f);
     //uint8_t stencil = depthBytes[3];
 #endif
-
+
```
Comment (samsinsane) [inline src/vcRender.cpp]: Whitespace here, but probably not worth delaying the merge of this.
